tests/unit/s2n_cipher_preference_test.c

```diff
@@ -14,8 +14,13 @@
  */
```

Comment:
Can there be a couple tests added to be sure this behavior is respected in the client_hello and server_hello?

Reply:
Sure testings on server_hello is added here.

For client hello, my code change only affects s2n_client_hello_recv function, not s2n_client_hello_send. And s2n_client_hello_recv is already tested within fuzz testings, which already includes versioning validation.